src/include/mempool.h

```diff
@@ -307,6 +307,33 @@ class pool_allocator {
     delete[] reinterpret_cast<char*>(p);
   }
 
+  T* allocate_aligned(size_t n, size_t align, void *p = nullptr) {
```

Comment:
conceptually, we would be better off implementing operator new[] with the new std::align_type_t (it's in C++17) to control the alignment -- right? But I don't think it's terribly important.